lib/util/hud_utility_2024.rb

```diff
@@ -0,0 +1,348 @@
+###
+# Copyright 2016 - 2023 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module HudUtility2024
+  include ::Concerns::HudLists2024
+  include ::Concerns::HudValidationUtil
+
+  module_function
+
+  # def race(field, reverse = false, multi_racial: false)
+  #   map = races(multi_racial: multi_racial)
+  #   _translate map, field, reverse
+  # end
+
+  # # NOTE: HUD, in the APR specifies these by order ID, as noted in the comments below
+  # def races(multi_racial: false)
+  #   race_list = {
+  #     'AmIndAKNative' => 'American Indian, Alaska Native, or Indigenous', # 1
+  #     'Asian' => 'Asian or Asian American', # 2
+  #     'BlackAfAmerican' => 'Black, African American, or African', # 3
+  #     'NativeHIPacific' => 'Native Hawaiian or Pacific Islander', # 4
+  #     'White' => 'White', # 5
+  #     'RaceNone' => 'Doesn\'t Know, refused, or not collected', # 6 (can be 99, 8, 9, null only if all other race fields are 99 or 0)
+  #   }
+  #   race_list['MultiRacial'] = 'Multi-Racial' if multi_racial
+  #   race_list
+  # end
+
+  # 1.6
+  def gender_none(id, reverse = false)
+    race_none(id, reverse)
+  end
+
+  def veteran_status(*args)
+    no_yes_reasons_for_missing_data(*args)
+  end
+
+  def project_type_number(type)
+    # attempt to lookup full name
+    number = project_type(type, true) # reversed
+    return number if number.present?
+
+    # perform an acronym lookup
+    project_type_brief(type, true) # reversed
+  end
+
+  def homeless_project_type_numbers
+    [
+      0, # ES E/E
+      1, # ES NBN
+      2, # TH
+      4, # SO
+      8, # SH
+    ].freeze
+  end
+
+  def gender_fields
+    gender_id_to_field_name.values.uniq.freeze
+  end
+
+  def gender_field_name_to_id
+    gender_id_to_field_name.invert.freeze
+  end
+
+  def gender_id_to_field_name
+    {
+      0 => :Woman,
+      1 => :Man,
+      2 => :CulturallySpecific,
+      3 => :DifferentIdentity,
+      4 => :NonBinary,
+      5 => :Transgender,
+      6 => :Questioning,
+      8 => :GenderNone,
+      9 => :GenderNone,
+      99 => :GenderNone,
+    }.freeze
+  end
+
+  def gender_comparison_value(key)
+    return key if key.in?([8, 9, 99])
+
+    1
+  end
+
+  # FIXME update
+  def no_single_gender_queries
+    [
+      '0,1',
+      '0,1,4',
+      '0,1,4,5',
+      '0,1,5',
+      '0,4',
+      '0,4,5',
+      '1,4',
+      '1,4,5',
+      '4',
+      '4,5',
+    ]
+  end
+
+  # FIXME update
+  def questioning_gender_queries
+    [
+      '0,1,4,5,6',
+      '0,1,4,6',
+      '0,1,5,6',
+      '0,1,6',
+      '0,4,5,6',
+      '0,4,6',
+      '0,5,6',
+      '0,6',
+      '1,4,5,6',
+      '1,4,6',
+      '1,5,6',
+      '1,6',
+      '4,5,6',
+      '4,6',
+      '5,6',
+      '6',
+    ]
+  end
+
+  # FIXME update
+  def transgender_gender_queries
+    [
+      '0,5',
+      '1,5',
+      '5',
+    ]
+  end
```

Comment:
I haven't updated these yet. It looks like they are possible values for the gender_multi column on some apr table. Is it just all possible sorted permutations of genders that is NOT 0 or 1? If so probably want to generate these rather than hard coding them

Reply:
These get used in the APR/CAPER/CE APR.  Very specifically defined there, and probably are changing, so leaving them as-is is correct for now.